app/src/main/java/com/example/albumsapp/SelectedPhotoFragment.kt

```diff
@@ -5,16 +5,41 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.navArgs
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.albumsapp.adapter.PhotoAdapter
+import com.example.albumsapp.databinding.FragmentSelectedPhotoBinding
+import com.example.albumsapp.viewmodel.PhotoViewModel
 
 class SelectedPhotoFragment : Fragment() {
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
+    private var _binding: FragmentSelectedPhotoBinding? = null
+    private val binding get() = _binding!!
+
+    private val args: SelectedPhotoFragmentArgs by navArgs()
+
+    private val viewModel: PhotoViewModel by viewModels {
+        PhotoViewModel.MyViewModelFactory(args.albumId)
     }
+
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        return inflater.inflate(R.layout.fragment_selected_photo, container, false)
+        _binding = FragmentSelectedPhotoBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        binding.currentPhoto.layoutManager = LinearLayoutManager(context)
+        viewModel.photos.observe(viewLifecycleOwner) {
+            binding.currentPhoto.adapter = PhotoAdapter(it[args.photo - 1])
```

Comment:
for the single photo screen we don't need a RecyclerView. We can just have ImageView for a photo and TextView for a title in ConstraintLayout or LinearLayout. RecyclerView is used only for a large number of similar elements in list or grids.

Reply:
Also it's a bit not optimized to download all photos in album here and then get only one we need.  Can we just download the single photo that we need?

Reply:
Also probably you are mixing ids and indexes. `args.photo` is `id` but you treat it as index here